src/filters/ws.rs

```diff
@@ -284,19 +185,26 @@ impl Stream for WebSocket {
     type Item = Result<Message, crate::Error>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
-        match (*self).with_context(Some(cx), |s| s.read_message()) {
-            Ok(item) => Poll::Ready(Some(Ok(Message { inner: item }))),
-            Err(::tungstenite::Error::Io(ref err)) if err.kind() == io::ErrorKind::WouldBlock => {
+        match Pin::new(&mut self.inner).poll_next(cx) {
+            Poll::Ready(Some(Ok(item))) => Poll::Ready(Some(Ok(Message { inner: item }))),
+            Poll::Ready(Some(Err(tungstenite::Error::Io(ref err))))
+                if err.kind() == io::ErrorKind::WouldBlock =>
```

Comment:
there were two, this one is still there

Reply:
Oh the `WouldBlock`, sorry, I thoght you meant the `with_context`.
It's not checked in the `tokio-tungstenite`'s [ `poll_next`](https://github.com/snapview/tokio-tungstenite/blob/607b8bf3d801bb14898fe4c5505cc6292950cd7a/src/lib.rs#L260), do I remove it anyway?

Reply:
no worries :) 
yeah, from what I see [poll_next](https://github.com/snapview/tokio-tungstenite/blob/607b8bf3d801bb14898fe4c5505cc6292950cd7a/src/lib.rs#L260) calls [cvt](https://github.com/snapview/tokio-tungstenite/blob/master/src/lib.rs#L268) under `with_context` that [does that conversion](https://github.com/snapview/tokio-tungstenite/blob/master/src/compat.rs#L200)

Reply:
Looking at `poll_next` I noticed that `Error::ConnectionClosed` is also already handled.

Reply:
Fixed, I removed the both the `WouldBlock` and the `ConnectionClosed` check

Reply:
thanks! had missed `Error::ConnectionClosed`.
 only some final minor nit that I also had missed, can you call [`ready!`](https://docs.rs/futures/0.3.1/futures/macro.ready.html) on the `Pin::new.poll` cases, so that Poll::Pending is handled, and the match brace becomes more readable, you have an example here: https://github.com/seanmonstar/warp/blob/master/src/filters/fs.rs#L399

Reply:
Done!